utils/ls8usgsprepare.py

```diff
@@ -0,0 +1,177 @@
+"""
+Ingest data from the command-line.
+"""
+from __future__ import absolute_import
+
+import uuid
+import logging
+import yaml
+import re
+import click
+from osgeo import osr
+import os
+
+try:
+    from urllib.request import urlopen
+    from urllib.parse import urlparse, urljoin
+except ImportError:
+    from urlparse import urlparse, urljoin
+    from urllib2 import urlopen
+
+MTL_PAIRS_RE = re.compile(r'(\w+)\s=\s(.*)')
+
+
+def _parse_value(s):
+    s = s.strip('"')
+    for parser in [int, float]:
+        try:
+            return parser(s)
+        except ValueError:
+            pass
+    return s
+
+
+def _parse_group(lines):
+    tree = {}
+
+    for line in lines:
+
+        match = MTL_PAIRS_RE.findall(line.decode('utf-8'))
+        if match:
+            key, value = match[0]
+            if key == 'GROUP':
+                tree[value] = _parse_group(lines)
+            elif key == 'END_GROUP':
+                break
+            else:
+                tree[key] = _parse_value(value)
+    return tree
+
+
+def get_geo_ref_points(info):
+    return {
+        'ul': {'x': info['CORNER_UL_PROJECTION_X_PRODUCT'], 'y': info['CORNER_UL_PROJECTION_Y_PRODUCT']},
+        'ur': {'x': info['CORNER_UR_PROJECTION_X_PRODUCT'], 'y': info['CORNER_UR_PROJECTION_Y_PRODUCT']},
+        'll': {'x': info['CORNER_LL_PROJECTION_X_PRODUCT'], 'y': info['CORNER_LL_PROJECTION_Y_PRODUCT']},
+        'lr': {'x': info['CORNER_LR_PROJECTION_X_PRODUCT'], 'y': info['CORNER_LR_PROJECTION_Y_PRODUCT']},
+    }
+
+
+def get_coords(geo_ref_points, spatial_ref):
+    t = osr.CoordinateTransformation(spatial_ref, spatial_ref.CloneGeogCS())
+
+    def transform(p):
+        lon, lat, z = t.TransformPoint(p['x'], p['y'])
+        return {'lon': lon, 'lat': lat}
+
+    return {key: transform(p) for key, p in geo_ref_points.items()}
+
+
+def get_mtl(path):
+    """
+    Path is pointing to the folder , where the USGS Landsat scene list in MTL format is downloaded
+    from Earth Explorer or GloVis
+    """
+    newfile = "Empty File"
+    for file in os.listdir(path):
+        if file.endswith("MTL.txt"):
+            metafile = file
+            newfile = open(os.path.join(path, metafile), 'rb')
+    return _parse_group(newfile)['L1_METADATA_FILE']
+
+
+#    return _parse_group(path)['L1_METADATA_FILE'])
+
+def prepare_dataset(path):
+    info = get_mtl(path)
+
+    level = info['PRODUCT_METADATA']['DATA_TYPE']
```

Comment:
It's probably coming from previous code, but nevertheless

I noticed that

```python
info['PRODUCT_METADATA']['SOME_KEY']
```
 is used in a lot of places.

It would be more readable to extract `info['PRODUCT_METADATA']` into a variable first.